src/framebuffer.ml

```diff
@@ -77,53 +78,46 @@ let draw_line (x0 : int) (y0 : int) (x1 : int) (y1 : int) (col : int) (buffer :
   and dy = (abs (y1 - y0)) * -1
   and sy = if y0 < y1 then 1 else -1 in
   let initial_error = dx + dy in
-
   let rec loop (x : int) (y : int) (error : int) =
-    if (x >= 0) && (x < Array.length (buffer.(0))) && (y >= 0) && (y < Array.length buffer) then
-      buffer.(y).(x) <- col;
-    match (x == x1) && (y == y1) with
-    | true -> ()
-    | false -> (
+    pixel_write x y col buffer;
+    if (x = x1) && (y = y1) then ()
```

Comment:
You changed the type of equality check here. Can you explain why? 

(I'm not saying it's wrong, but if you change code you should be able to to say what was wrong before, and why you picked the new version)